subt/docker/robotika/ros_proxy_node.cc

```diff
@@ -0,0 +1,531 @@
+             pose.orientation.x, pose.orientation.y, pose.orientation.z, pose.orientation.w);
+
+  // Simple example for robot to go to entrance
+  geometry_msgs::Twist msg;
+  while(this->getSpeedCmd(msg))
+  {
+    this->velPub.publish(msg);
+  }
+
+  // Distance to goal
+  double dist = pose.position.x * pose.position.x +
+    pose.position.y * pose.position.y;
+
+  if(abs(prev_dist2 - dist) > 1.0)
+  {
+    ROS_INFO_STREAM("MD robot pose " << pose.position.x << " " << pose.position.y << " dist=" << dist);
+    prev_dist2 = dist;
+  }
+
+  // Arrived
+  if (dist < 0.3 || pose.position.x >= -0.3)
+  {
+    this->arrived = true;
+    ROS_INFO("Arrived at entrance!");
+  }
+}
+
+
+/////////////////////////////////////////////////
+bool Controller::parseArtf(char *input_str, subt::msgs::Artifact& artifact)
+{
+  double x, y, z;
+  char buf[256];
+
+  if(sscanf(input_str, "%s %lf %lf %lf", buf, &x, &y, &z) == 4)
+  {
+        ROS_INFO_STREAM("MD artf = " << buf);
+        ROS_INFO_STREAM("MD x = " << x);
+        ROS_INFO_STREAM("MD y = " << y);
+        ROS_INFO_STREAM("MD z = " << z);
+
+        artifact.mutable_pose()->mutable_position()->set_x(x);
+        artifact.mutable_pose()->mutable_position()->set_y(y);
+        artifact.mutable_pose()->mutable_position()->set_z(z);
+
+        subt::ArtifactType type;
+        if(strcmp(buf, "TYPE_BACKPACK") == 0)
+        {
+          type = subt::ArtifactType::TYPE_BACKPACK;
+        }
+        if(strcmp(buf, "TYPE_DUCT") == 0)
+        {
+          type = subt::ArtifactType::TYPE_DUCT;
+        }
+        if(strcmp(buf, "TYPE_DRILL") == 0)
+        {
+          type = subt::ArtifactType::TYPE_DRILL;
+        }
+        if(strcmp(buf, "TYPE_ELECTRICAL_BOX") == 0)
+        {
+          type = subt::ArtifactType::TYPE_ELECTRICAL_BOX;
+        }
+        if(strcmp(buf, "TYPE_EXTINGUISHER") == 0)
+        {
+          type = subt::ArtifactType::TYPE_EXTINGUISHER;
+        }
+        if(strcmp(buf, "TYPE_PHONE") == 0)
+        {
+          type = subt::ArtifactType::TYPE_PHONE;
+        }
+        if(strcmp(buf, "TYPE_RADIO") == 0)
+        {
+          type = subt::ArtifactType::TYPE_RADIO;
+        }
+        if(strcmp(buf, "TYPE_RESCUE_RANDY") == 0)
+        {
+          type = subt::ArtifactType::TYPE_RESCUE_RANDY;
+        }
+        if(strcmp(buf, "TYPE_TOOLBOX") == 0)
+        {
+          type = subt::ArtifactType::TYPE_TOOLBOX;
+        }
+        if(strcmp(buf, "TYPE_VALVE") == 0)
+        {
+          type = subt::ArtifactType::TYPE_VALVE;
+        }
+
+        artifact.set_type(static_cast<uint32_t>(type));
+
+        ROS_INFO_STREAM("MD enum = " << static_cast<uint32_t>(type));
+        return true;
+  }
+  return false;
+}
+
+/////////////////////////////////////////////////
+bool Controller::getSpeedCmd(geometry_msgs::Twist& msg)
+{ 
+  char buffer[10000];
+  int size;
+  while((size=zmq_recv(g_requester, buffer, 10000, ZMQ_DONTWAIT)) > 0)
```

Comment:
I don't recognize the signature of [zmq_recv](http://api.zeromq.org/2-1:zmq-recv). Are you using something else?